mapie_v1/utils.py

```diff
@@ -0,0 +1,271 @@
+    else:
+        confidence_levels = [confidence_level]
+    return [
+        transform_confidence_level_to_alpha(confidence_level)
+        for confidence_level in confidence_levels
+    ]
+
+
+# Could be replaced by using sklearn _validate_params (ideally wrapping it)
+def check_if_param_in_allowed_values(
+    param: str, param_name: str, allowed_values: list
+) -> None:
+    if param not in allowed_values:
+        raise ValueError(
+            f"'{param}' option not valid for parameter '{param_name}'"
+            f"Available options are: {allowed_values}"
+        )
+
+
+def check_cv_not_string(cv: Union[int, str, BaseCrossValidator]) -> None:
+    if isinstance(cv, str):
+        raise ValueError(
+            "'cv' string options not available in MAPIE >= v1.0.0"
+        )
+
+
+def cast_point_predictions_to_ndarray(
+    point_predictions: Union[NDArray, Tuple[NDArray, NDArray]]
+) -> NDArray:
+    if isinstance(point_predictions, tuple):
+        raise TypeError(
+            "Developer error: use this function to cast point predictions only, "
+            "not points + intervals."
+        )
+    return cast(NDArray, point_predictions)
+
+
+def cast_predictions_to_ndarray_tuple(
+    predictions: Union[NDArray, Tuple[NDArray, NDArray]]
+) -> Tuple[NDArray, NDArray]:
+    if not isinstance(predictions, tuple):
+        raise TypeError(
+            "Developer error: use this function to cast predictions containing points "
+            "and intervals, not points only."
+        )
+    return cast(Tuple[NDArray, NDArray], predictions)
+
+
+def prepare_params(params: Union[dict, None]) -> dict:
+    return copy.deepcopy(params) if params else {}
+
+
+def prepare_fit_params_and_sample_weight(
+    fit_params: Union[dict, None]
+) -> Tuple[dict, Optional[ArrayLike]]:
+    fit_params_ = prepare_params(fit_params)
+    sample_weight = fit_params_.pop("sample_weight", None)
+    return fit_params_, sample_weight
+
+
+def raise_error_if_previous_method_not_called(
+    current_method_name: str,
+    previous_method_name: str,
+    was_previous_method_called: bool,
+) -> None:
+    if not was_previous_method_called:
+        raise ValueError(
+            f"Incorrect method order: call {previous_method_name} "
+            f"before calling {current_method_name}."
+        )
+
+
+def raise_error_if_method_already_called(
+    method_name: str,
+    was_method_called: bool,
+) -> None:
+    if was_method_called:
+        raise ValueError(
+            f"{method_name} method already called. "
+            f"MAPIE does not currently support calling {method_name} several times."
+        )
+
+
+def raise_error_if_fit_called_in_prefit_mode(
+    is_mode_prefit: bool,
+) -> None:
+    if is_mode_prefit:
+        raise ValueError(
+            "The fit method must be skipped when the prefit parameter is set to True. "
+            "Use the conformalize method directly after instanciation."
+        )
+
+
+def train_conformalize_test_split(
```

Comment:
To put on top of the file (this is the only public utility for now, better to make it easily visible)